<launch>
    <param name="/use_sim_time" value="false" />

   <!-- Start the Dynamixel low-level driver manager with parameters -->
   <node name="makin_dynamixel_manager" pkg="dynamixel_controllers"
      type="controller_manager.py" required="true" output="screen">
      <rosparam>
         namespace: turtlebot_dynamixel_manager
         serial_ports:
            dynamixel_ax12:
               port_name: "/dev/ttyUSB0"
               baud_rate: 1000000
               min_motor_id:  1
               max_motor_id: 30
               update_rate: 20
      </rosparam>
   </node>
   
    <!-- Load the joint controller configuration from a YAML file -->
    <rosparam file="$(find slam_makin)/config/dynamixel_params.yaml" command="load"/>

   <!-- Start the head pan and tilt controllers -->
   <node name="dynamixel_controller_spawner_ax12" pkg="dynamixel_controllers"
   type="controller_spawner.py"
        args="--manager=turtlebot_dynamixel_manager
              --port=dynamixel_ax12
              --type=simple
        head_pan_joint"
        output="screen" />

   <!-- Start all Dynamixels in the relaxed state -->
   <!-- <node pkg="slam_makin" type="relax_all_servos.py" name="relax_all_servos" /> -->
   
</launch>
